main.js

```diff
@@ -274,6 +276,19 @@ function startApp() {
 
   }
 
+  function generateUUID() { // Public Domain/MIT
+    var d = new Date().getTime();
+    if (typeof performance !== 'undefined' && typeof performance.now === 'function'){
+        d += performance.now(); //use high-precision timer if available
+    }
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+        var r = (d + Math.random() * 16) % 16 | 0;
+        d = Math.floor(d / 16);
+        return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
+    });
+  }
```

Comment:
We already have a v4 UUID generator required in this process.  Is there a reason we can't use that?  Do these IDs need to conform to another standard?

Reply:
Yeah I know why the tests are failing, fixed a few already.

Oh yeah we can probably use it.